csrc/executor.h

```diff
@@ -248,6 +253,10 @@ class TORCH_CUDA_CU_API FusionExecutor : public NonCopyable {
   std::vector<at::Tensor> allocOutputSpace(
       const at::ArrayRef<c10::IValue>& inputs);
 
+  Fusion* fusion() const {
```

Comment:
From @naoyam 
Doesn't this return the same as kernel()? Perhaps, we should remove fusion_ and replace it with kernel().

Reply:
nitpick: a const overload.
Or rather, is this one supposed to return a mutable fusion?! That doesn't sound right, maybe just return const.

Reply:
I think I walked into something complicated...

@naoyam is right than I can simple use `kernel()->as<Fusion>()` instead of adding a new method to `FusionExecutor`.  Therefore, I add that, for now.

@jjsjann123 It is also true, that from a usage perspective, I am inspecting the fusion but not changing it which implies treating it as `const` could be desirable for protecting my intention via the compiler.  However, if I return a `const Fusion*` that could mean I need to do one of three things:
1. I would need to `const_cast` away the prior `const` definition of the returned `Fusion` object prior to inserting it into the `FusionGuard`.
2. Add explicit `const` methods to the `FusionGuard` and perhaps a `const` static data member.
3. Create a templated version of the `FusionGuard` that creates instances of a mutable guard and a const guard.

Adding `const Fusion` object pointers seems more complicated than I expected.  Therefore, I am wondering if that is something I should really do in this PR?  I have done what Naoya suggested in using `FusionExecutor::kernel()`, for now, as it is the least complicated change.

Reply:
ha, it's always `FusionGuard` that got me by surprise. :crying_cat_face:  Sorry for the distraction here, that sounds like a rabbit hole.
I do think we need to revisit the const correctness of our design around Fusion/FusionGuard, but let's not put it in this PR yet.